src/server/keystore/keystore.js

```diff
@@ -0,0 +1,103 @@
+import { writeFileSync, readFileSync, existsSync } from 'fs';
+import { createCipheriv, createDecipheriv, randomBytes, pbkdf2Sync } from 'crypto';
+import * as errors from './errors';
+
+const VERSION = 1;
+const ALGORITHM = 'aes-256-gcm';
+const ITERATIONS = 250;
+
+export class Keystore {
+  constructor(path, password = '') {
+    this.path = path;
+    this.data = {};
+    this.password = password;
+
+    this.load();
+  }
+
+  static errors = errors;
+
+  static encrypt(text, password = '') {
+    const iv = randomBytes(12);
+    const salt = randomBytes(64);
+    const key = pbkdf2Sync(password, salt, ITERATIONS, 32, 'sha512');
+
+    const cipher = createCipheriv(ALGORITHM, key, iv);
+
+    const ciphertext = Buffer.concat([cipher.update(text, 'utf8'), cipher.final()]);
+    const tag = cipher.getAuthTag();
+
+    return Buffer.concat([salt, iv, tag, ciphertext]).toString('base64');
+  }
+
+  static decrypt(data, password = '') {
+    try {
+      const bData = new Buffer(data, 'base64');
+
+      // convert data to buffers
+      const salt = bData.slice(0, 64);
+      const iv = bData.slice(64, 76);
+      const tag = bData.slice(76, 92);
+      const text = bData.slice(92);
+
+      const key = pbkdf2Sync(password, salt, ITERATIONS, 32, 'sha512');
```

Comment:
For the number of iterations, where did the value come from? I think we probably need to use more iterations. NIST's guidelines recommend 10000 iterations. https://pages.nist.gov/800-63-3/sp800-63b.html#sec5

> For PBKDF2, the cost factor is an iteration count: the more times the PBKDF2 function is iterated, the longer it takes to compute the password hash. Therefore, the iteration count SHOULD be as large as verification server performance will allow, typically at least 10,000 iterations.

Reply:
Are there any plans to make this configurable?  I'd have np making this 100k+ in production(but not for dev)

Reply:
The iterations have been bumped to 10,000.

https://github.com/elastic/kibana/pull/14714/commits/84041bb9096b1f06cbbc8a914b12fb89bbba702b#diff-ce5a54d0ba6d5b17313dd55267cfe902R7